src/config.jl

```diff
@@ -11,12 +11,11 @@ let
     global mysql_lib
     succeeded = false
     if !isdefined(:mysql_lib)
-        @linux_only lib_choices = ["libmysql.so", "libmysqlclient.so",
-                                   "libmysqlclient_r.so", "libmariadb.so",
-                                   "libmysqlclient_r.so.16"]
-        @osx_only lib_choices = ["libmysqlclient.dylib"]
-        @windows_only lib_choices = ["libmysql.dll", "libmariadb.dll"]
-        local lib
```

Comment:
all of this logic should be happening at build time in a `deps/build.jl` and the results cached rather than happening at load time in `src`...